### PR TITLE
feat(metrics-extraction): Set spec limit with stateful extraction

### PR DESCRIPTION
**Pull request by k-fish**
### Summary

This re-arranges widgets to exclude only the least recently visited widget, and in the case that you're over your limit it'll mark down which ones were excluded in the extraction state table. This way once we've passed the org one time we should fill out the spec-limit and maintain a stable set of specs across time moving forward.

### CODE REVIEW

File: src/sentry/relay/config/metric_extraction.py [hunk 1]
```diff
@@ -291,10 +306,40 @@ def _trim_if_above_limit(
                 )
 
             return_specs += specs_for_version[:max_specs]
+            if (
+                OnDemandMetricSpecVersioning.get_query_spec_version(project.organization.id)
+                == SpecVersion(1)
```
Comment (armenzg) [inline src/sentry/relay/config/metric_extraction.py:311]: What's the purpose of this check?

You may want to do `OnDemandMetricSpecVersioning.spec_versions[0]` or `OnDemandMetricSpecVersioning.spec_versions[-1]` depending if you want the current version or the latest version.

File: src/sentry/relay/config/metric_extraction.py [hunk 2]
```diff
@@ -270,9 +284,10 @@ def _trim_if_above_limit(
     max_specs: int,
     project: Project,
     widget_type: str,
-) -> list[HashedMetricSpec]:
+) -> Tuple[list[HashedMetricSpec], list[HashedMetricSpec]]:
     """Trim specs per version if above max limit"""
```
Comment (armenzg) [inline src/sentry/relay/config/metric_extraction.py:288]: Please update the docstring.

**Review by armenzg — APPROVED** [review]
Looks good to me 👍🏻

File: src/sentry/relay/config/metric_extraction.py (hunk 1 above)
Comment (k-fish) [reply src/sentry/relay/config/metric_extraction.py]: We don't need it now that there are multiple spec rows.

**Pull request merged**